File: Assign2/RD_JAVA/States.java
package RD_JAVA;


public class States {
    String trimFormula; // 수식
    int currentIndex; // 현재 인덱스
    int countParens; // 괄호 카운트

    public States(String formula) {
        trimFormula = formula.trim().replace(" ", "");
        currentIndex = 0;
        countParens = 0;
    }

    public void nextToken() {
        if (currentIndex < trimFormula.length()) {
            char ch = trimFormula.charAt(currentIndex);
            if (ch == '(') {
                countParens++;
            } else if (ch == ')') {
                countParens--;
            }
            currentIndex++;
        }
    }

    public char getToken() {
        if (currentIndex < trimFormula.length()) {
            return trimFormula.charAt(currentIndex);
        } else {
            return 0;
        }
    }
}

File: RD_Parser/Python/States.py

class States:

    def __init__(self, formula):
        self.trim_Formula = formula
        self.current_Index = 0
        self.count_Parens = 0

    def nextToken(self):
        if self.current_Index < len(self.trim_Formula):
            ch = self.trim_Formula[self.current_Index]
            if ch == "(":
                self.count_Parens += 1
            elif ch == ')':
                self.count_Parens -= 1
            self.current_Index += 1

    def getToken(self):
        if self.current_Index < len(self.trim_Formula):
            return self.trim_Formula[self.current_Index]
        else:
            return 0




